### PR TITLE
Add layer normalization as an option for residual layer in TCN

### PR DESCRIPTION
**Pull request by adamconkey**
Layer Normalization is implemented in the TensorFlow Keras API and has the same usage as the Batch Normalization layer currently being used in the residual layer of the TCN. I've added an option to use LayerNorm instead of BatchNorm as Layer Normalization generally has known benefits over Batch Normalization and it's nice to have the option.

### CODE REVIEW

**Comment by adamconkey** [comment]
I used an `else if` because I doubt anyone wants to use *both* BatchNorm and LayerNorm in the same layer, but I could be wrong. If so, can change `else if` to `if`. It may also be good to error check and error out or warn the user if they set both `use_batch_norm` and `use_layer_norm` to be True.

**Comment by philipperemy** [comment]
@adamconkey great work! We def need to merge this one :)

**Comment by philipperemy** [comment]
Yes I agree it would sound weird for users to input both `use_batch_norm` and `use_layer_norm` at the same time. An error seems legit.
